### ISSUE TITLE
Generalise demosaicing physics >3ch

### ISSUE THREAD

**Issue by Andrewwango** [deepinv/deepinv#1329]
### Describe the feature

`deepinv.physics.Demosaicing` is a thin wrapper over `Inpainting` using a mask which represents the 3 channel Bayer filter. We should allow instantiating Demosaicing operators with other common number of channels for snapshot multispectral cameras using various common multispectral filter arrays.

### Motivation

In #1311 , the demosaicing example's physics could be more cleanly instantiated e.g.:
```python
physics = Demosaicing(img_size=(16, ..., ...))
```
rather than what it is now
```python
mask = torch.zeros_like(y)
for i in range(4):
    for j in range(4):
        c = i * 4 + j
        mask[:, c, i::4, j::4] = 1

physics = dinv.physics.Inpainting(img_size=mask.shape[1:], mask=mask, device=device)
```
